lms/djangoapps/instructor_task/tests/test_integration.py

```diff
@@ -396,6 +424,72 @@ def test_rescoring_randomized_problem(self):
         for user in self.users:
             self.check_state(user, descriptor, 0, 1, expected_attempts=2)
 
+    @patch('lms.djangoapps.instructor_task.tasks_helper.tracker')
+    @patch('lms.djangoapps.grades.signals.handlers.tracker')
+    @patch('lms.djangoapps.grades.models.tracker')
+    def test_rescoring_events(self, grades_tracker, handlers_tracker, instructor_task_tracker):
+        problem_edit = dict(correct_answer=OPTION_2)
+        self.verify_rescore_for_one_student(
+            problem_edit, 0, 2, rescore_if_higher=False,
+        )
+        user_action_id = instructor_task_tracker.method_calls[0][1][1]['user_action_id']
+        calls_after_rescore = [
+            call for call in grades_tracker.method_calls
+            if call[1][1]['user_action_type'] == u'edx.grades.problem.rescored'
+        ]
+        # check that events caused by the rescore (and not the initial
+        # problem submission) have the expected id
+        for call in calls_after_rescore:
+            self.assertEqual(user_action_id, call[1][1]['user_action_id'])
+
+        instructor_task_tracker.emit.assert_called_with(
+            u'edx.grades.problem.rescored',
+            {
+                'course_id': unicode(self.course.id),
+                'user_id': unicode(self.user1.id),
+                'problem_id': unicode(InstructorTaskModuleTestCase.problem_location('H1P1')),
+                'original_weighted_earned': 2,
+                'new_weighted_earned': 0,
+                'original_weighted_possible': 2,
+                'new_weighted_possible': 2,
+                'only_if_higher': False,
+                'instructor_username': u'instructor',
+                'user_action_id': user_action_id,
+                'user_action_type': u'edx.grades.problem.rescored',
+            }
+        )
+
+        handlers_tracker.emit.assert_called_with(
+            u'edx.grades.problem.submitted',
```

Comment:
I don't disagree. Should we skip emitting this problem if the `event_tranasaction_type` is set- that is to say, if we have reached this point in the code via something other than a student submitting a problem attempt?